h5py/tests/common.py

```diff
@@ -165,3 +167,32 @@ def closed_tempfile(suffix='', text=None):
             test_file.flush()
     yield file_name
     shutil.rmtree(file_name, ignore_errors=True)
+
+
+def insubprocess(f):
+    """Runs a test in its own subprocess"""
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        curr_test = os.environ.get('PYTEST_CURRENT_TEST', None)
```

Comment:
This looks convenient, but the pytest docs explicitly say you shouldn't rely on the format:

> The contents of PYTEST_CURRENT_TEST is meant to be human readable and the actual format can be changed between releases (even bug fixes) so it shouldn’t be relied on for scripting or automation.

https://docs.pytest.org/en/latest/example/simple.html#pytest-current-test-env

It looks like the request object might be provide a way to get the necessary information: https://stackoverflow.com/a/34732269/434217 . I don't know how easily you can get that in a decorator.

Reply:
updated!